core/src/main/java/com/linecorp/armeria/internal/server/annotation/AnnotatedService.java

```diff
@@ -324,11 +331,16 @@ HttpService withExceptionHandler(HttpService service) {
      * {@link HttpResponse}, it will be executed in the blocking task executor.
      */
     private CompletionStage<HttpResponse> serve0(ServiceRequestContext ctx, HttpRequest req) {
-        final CompletableFuture<AggregatedHttpRequest> f;
+        final CompletableFuture<AggregatedResult> f;
         if (AggregationStrategy.aggregationRequired(aggregationStrategy, req)) {
-            f = req.aggregate();
+            final MediaType contentType = req.contentType();
+            if (contentType != null && contentType.isMultipart()) {
+                f = aggregateMultipart(ctx, req).thenApply(AggregatedResult::new);
+            } else {
+                f = req.aggregate().thenApply(AggregatedResult::new);
+            }
```

Comment:
This conflicts with `ExchangeType` PR. https://github.com/line/armeria/pull/3956/files#diff-a5b50e72b0031befe6de444cf1fc3d47c573cd73a65312068dd7ba99d440f8b1.
If `AggregationStrategy.aggregationRequired(aggregationStrategy, req)` returns true, a request will be aggregated at `Http{1,2}RequestDecoder` level.
Could we migrate `AggregationStrategy.aggregationRequired(...)` to `AggregationStrategy.aggregationType(...)`?

```java
// Not sure this is good.
// You can ignore this type and naming suggestion.
enum AggregationType {
    ALL,
    MULTIPART,
    NONE,
}
```

Reply:
so if I add AggregationType maybe the ExchangeType PR's `public ExchangeType exchangeType(RequestHeaders headers, Route route) {` need to change both aggregationRequired & AggregationType ?

Reply:
Have migrated to `AggregationType`. LGTM so far.
Only one concern is currently resolver & annotated service decide the strategy/type separately. For current implementations it's enough. But if we want to support `AggregatedMultipart`(I didn't), we need to make resolver also know it wants to aggregate in memory or file first? Also `AggregatedResult` may need new field `AggregatedMultipart`. Currently I give up to support `AggregatedMultipart` and think user can do similar thing by using `Multipart#aggregate` by self.